airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py

```diff
@@ -222,31 +206,37 @@ def parse_response(self, response: requests.Response, stream_state: Mapping[str,
                 yield question
 
 
-class SurveyResponses(IncrementalSurveymonkeyStream):
+class SurveyResponses(SliceMixin, IncrementalSurveymonkeyStream):
     """
     Docs: https://developer.surveymonkey.com/api/v3/#survey-responses
     """
 
     cursor_field = "date_modified"
 
     def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
-        survey_id = stream_slice["survey_id"]
-        return f"surveys/{survey_id}/responses/bulk"
-
-    def stream_slices(self, **kwargs):
-        survey_stream = Surveys(start_date=self._start_date, survey_ids=self._survey_ids, authenticator=self.authenticator)
-        for survey in survey_stream.read_records(sync_mode=SyncMode.full_refresh):
-            yield {"survey_id": survey["id"]}
+        return f"surveys/{stream_slice['survey_id']}/responses/bulk"
 
     def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
         """
         Return the latest state by comparing the survey_id and cursor value in the latest record with the stream's most recent state object
         and returning an updated state object.
         """
         survey_id = latest_record.get("survey_id")
-        return {
```

Comment:
inital code was incorrect - the expected state format is 

we need to save format for each survey id. like
{ <survey_id>: {<cursot>: state} }